### PR TITLE
pkg/trace/api: Move semantic conventions to separate internal package

### PR DESCRIPTION
**Pull request by mx-psi**
### What does this PR do?

Moves header and tag definitions to separate internal package. This is a refactor with no end-user visible changes.

### Motivation

Expose dependencies of Trace Agent OTLP receiver. The end goal is to make this an independent component on a separate Go module.

### Additional Notes

I chose the name `semconv` (semantic conventions) which is the one use in the OpenTelemetry world for 'strings with a specific semantic meaning'. I am happy to change the name to something else if someone comes up with a better name.

### Reviewer's Checklist

- [ ] If known, an appropriate milestone has been selected; otherwise the `Triage` milestone is set.
- [ ] Use the `major_change` label if your change either has a major impact on the code base, is impacting multiple teams or is changing important well-established internals of the Agent. This label will be use during QA to make sure each team pay extra attention to the changed behavior. For any customer facing change use a releasenote.
- [ ] A [release note](https://github.com/DataDog/datadog-agent/blob/main/docs/dev/contributing.md#reno) has been added or the `changelog/no-changelog` label has been applied.
- [ ] Changed code has automated tests for its functionality.
- [ ] Adequate QA/testing plan information is provided if the `qa/skip-qa` label is not applied.
- [ ] At least one `team/..` label has been applied, indicating the team(s) that should QA this change.
- [ ] If applicable, docs team has been notified or [an issue has been opened on the documentation repo](https://github.com/DataDog/documentation/issues/new).
- [ ] If applicable, the `need-change/operator` and `need-change/helm` labels have been applied.
- [ ] If applicable, the `k8s/<min-version>` label, indicating the lowest Kubernetes version compatible with this feature. 
- [ ] If applicable, the [config template](https://github.com/DataDog/datadog-agent/blob/main/pkg/config/config_template.yaml) has been updated.

### CODE REVIEW

**Review by gbbr — COMMENTED** [review]
1. Is this an intermediate step so that both the `pkg/trace/api` package and the OTLP can use it? If so, how will the new OTEL code be able to do that if it's internal?
2. I would use something different than `semconv`. It will just end up clashing with the OTEL `semconv` package and we'll have to keep aliasing.

**Comment by mx-psi** [comment]
Renamed to `shared` per offline conversation.

File: pkg/trace/api/internal/shared/conventions.go [hunk 1]
```diff
@@ -0,0 +1,61 @@
...
+	// This value is used for metrics and could be used in the future to adjust priority rates.
+	HeaderDroppedP0Spans = "Datadog-Client-Dropped-P0-Spans"
+
+	// HeaderRatesPayloadVersion contains the version of sampling rates.
+	// If both agent and client have the same version, the agent won't return rates in API response.
+	HeaderRatesPayloadVersion = "Datadog-Rates-Payload-Version"
+
+	// TagContainersTags specifies the name of the tag which holds key/value
+	// pairs representing information about the container (Docker, EC2, etc).
+	TagContainersTags = "_dd.tags.container"
```
Comment (gbbr) [inline pkg/trace/api/internal/shared/conventions.go:60]: How about we move this out of here? It's fine if we need to duplicate in two packages. It's quite common. It is already duplicated in multiple places anyway: backend, clients, agent. Then, we can keep all the rest if you like this approach and call the package `httpheader` or `header` and remove all the `Header*` prefixes from the constants.

WDYT?

Comment (mx-psi) [reply pkg/trace/api/internal/shared/conventions.go]: @gbbr Sounds reasonable to me, I renamed to `header`, PTAL

**Review by gbbr — APPROVED** [review]
LGTM!

**Pull request merged**